x-pack/plugins/fields_metadata/public/hooks/use_fields_metadata/use_fields_metadata.test.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
+
+import { createUseFieldsMetadataHook, UseFieldsMetadataParams } from './use_fields_metadata';
+import { FindFieldsMetadataResponsePayload } from '../../../common/latest';
+import { createFieldsMetadataServiceStartMock } from '../../services/fields_metadata/fields_metadata_service.mock';
+import { IFieldsMetadataClient } from '../../services/fields_metadata';
+
+const fields: FindFieldsMetadataResponsePayload['fields'] = {
+  '@timestamp': {
+    dashed_name: 'timestamp',
+    description:
+      'Date/time when the event originated.\nThis is the date/time extracted from the event, typically representing when the event was generated by the source.\nIf the event source has no original timestamp, this value is typically populated by the first time the event was received by the pipeline.\nRequired field for all events.',
+    example: '2016-05-23T08:05:34.853Z',
+    flat_name: '@timestamp',
+    level: 'core',
+    name: '@timestamp',
+    normalize: [],
+    short: 'Date/time when the event originated.',
+    type: 'date',
+    source: 'ecs',
+  },
+};
+
+const mockedFieldsMetadataResponse = { fields };
+
+const fieldsMetadataService = createFieldsMetadataServiceStartMock();
+
+const useFieldsMetadata = createUseFieldsMetadataHook({ fieldsMetadataService });
+
+describe('useFieldsMetadata', () => {
+  let fieldsMetadataClient: jest.Mocked<IFieldsMetadataClient>;
+  beforeEach(async () => {
+    jest.clearAllMocks();
```

Comment:
nit: I always feel with this pattern, that we have a spillover somewhere. Can we add clearAllMocks in an afterEach instead here